core/src/main/java/org/eclipse/hono/util/CommandRouterConstants.java

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.util;
```

Comment:
is this class required to be in the core module? IMHO it will only ever be needed in the context of the client itself, or am I mistaken?

Reply:
There are 2 use cases of the class:
The client uses the endpoint name and the action names ("set-last-known-gw", etc.) when opening the link/sending messages. (And the endpoint name is used as a bean qualifier.)
The server uses the endpoint name and actions in `DelegatingCommandRouterAmqpEndpoint` to receive/map incoming messages. That class would go into the `service-base` module if sticking to current conventions.

So, if not wanting to duplicate endpoint name and actions, I don't see a different, common place where these things can be put in. The `service-base` module isn't an option because it isn't referenced by `hono-client-adapter-amqp`.

Reply:
I see. You're right ...